apis/v1alpha2/validation/grpcroute.go

```diff
@@ -17,11 +17,26 @@ limitations under the License.
 package validation
 
 import (
+	"fmt"
+	"net/http"
+	"strings"
+
 	"k8s.io/apimachinery/pkg/util/validation/field"
 
 	gatewayv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
 )
 
+var (
+	// repeatableGRPCRouteFilters are filter types that can are allowed to be
+	// repeated multiple times in a rule.
+	repeatableGRPCRouteFilters = []gatewayv1a2.GRPCRouteFilterType{
+		gatewayv1a2.GRPCRouteFilterExtensionRef,
+	}
+
+	invalidPathSequences = []string{"//", "/./", "/../", "%2f", "%2F", "#"}
+	invalidPathSuffixes  = []string{"/..", "/."}
```

Comment:
For future readers, it might be helpful to add comments to these explaining what, and maybe even some of the why.

Reply:
Added, thanks!